### PR TITLE
[ISV-4497] validate catalog format

### PR DESCRIPTION
**Pull request by mantomas**
JIRA: ISV-4497

### CODE REVIEW

File: operator-pipeline-images/operatorcert/entrypoints/validate_catalog_format.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+
+    return parser
+
+
+def run_makefile(makefile_path: str) -> None:
+    """Run makefile"""
+    cmd = [
+        "make",
```
Comment (Allda) [inline operator-pipeline-images/operatorcert/entrypoints/validate_catalog_format.py:50]: The makefile target should be `make catalog` based on a template given to us by opm team.

Comment (mantomas) [reply operator-pipeline-images/operatorcert/entrypoints/validate_catalog_format.py:50]: Changed, thanks for pointing out.

**Review by mavaras — APPROVED** [review]
LGTM :+1:

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml [hunk 2]
```diff
@@ -411,12 +411,36 @@ spec:
...
+        - name: pipeline_image
+          value: "$(params.pipeline_image)"
+        - name: affected_catalogs
+          value: "$(tasks.detect-changes.results.affected_catalogs)"
+        - name: operator_path
+          value: "$(tasks.detect-changes.results.operator_path)"
+      workspaces:
+        - name: source
+          workspace: repository
+          subPath: src
```
Comment (mporrato) [inline ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml:436]: We need to make sure the makefile cannot tamper with the files in the repository workspace.

**Review by mporrato — CHANGES_REQUESTED** [review]

**Comment by mantomas** [comment]
Removed Makefile validation and changed task order to first validate the catalog format and then build the fragment image. Test pipeline can be seen [here](https://gist.github.com/rh-operator-bundle-bot/835d3ac2a5935a567f2e7436c65739b8).

**Review by mporrato — APPROVED** [review]
Since this is now a rather trivial test, I would have done it as a static check in the common suite, but this approach should work too.

**Pull request merged**